src/rebar_otp_app.erl

```diff
@@ -156,6 +156,29 @@ load_app_vars(State) ->
             Vars
     end.
 
+get_app_vars_file_from_state(AppName, State) ->
+    case rebar_state:get(State, app_vars_file, undefined) of
```

Comment:
This is the wrong call to use I believe. The problem here is that this will use the top-level rebar.config file and inject its values into _all_ OTP Applications. I think it would make more sense to use `rebar_app_info:get` to fetch the values within the app's own rebar configuration value.